### PR TITLE
[Modular][Ready] Winterized Syndicate hardsuit conversion kit

### PR DESCRIPTION
**Pull request by Ebin-Halcyon**
## About The Pull Request

**HUGE THANK YOU TO @OrionTheFox FOR ACTUALLY MAKING THE RADIALS WORK. As the modification kit framework was merged, the work is now rendered moot, thank you for helping me through when it wasn't merged though. <3**

Adds in two new looking variants of the classic Blood-Red hardsuit, A slick white, black, red and fur lined variant of the Hardsuit. With and without tassets. Along with specialized kits with custom sprites made for them from #8506.

As of this PR the only way to get the hardsuit are from the kits or to have them specially admin spawned in. Interdyne/DS-2 get these kits from the get-go so they should be the only ones with them unless mappers add the kits to other areas.

Both plantigrade and Digitigrade have custom sprites
EVA Directionals:

Combat Directionals:

Naga sprites:

## How This Contributes To The Skyrat Roleplay Experience

A nicer looking variant of the original already nice looking hardsuit is pretty nice, and people might appreciate being able to run around in something other than the original redsuit for Interdyne/DS-2.
We also have actual winter gear to walk around in on an ice planet now as well.

## Changelog

:cl: The Syndicate
add: We have exported newly designed Winterized Hardsuits for frozen conditions, beware it does not have any better protection.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
why doesnt this just use the alternate skin system that like double barrel shotguns or captain pens have

**Comment by YakumoChen** [comment]
not really a fan? The hardsuits are iconic. I feel like its too easy to confuse them with the other white hardsuits we currently have (especially the security one that doesn't see action most rounds)

Not to mention they aren't... y'know... _blood red_ anywhere. They _need_ to be iconic as the crew is trained to spot the originals and almost everyone who plays ss13 knows what a blood-red hardsuit normally looks like.

**Comment by bluetshirtguy** [comment]
They already decided to differentiate sec from the syndicate with red and blue, what justifies changing the syndicate red and black to a grey, white and black?

**Comment by Ebin-Halcyon** [comment]
> why doesnt this just use the alternate skin system that like double barrel shotguns or captain pens have

Tried doing it before with Orion but couldn't get it to work, along with alt-click already being the 'open the cell cover' hotkey I'm a shoddy at best coder and wouldn't be able to make a work around or stuff in already existing code and make it work. Everyone that could help is a bit busy with making the new prefs menu work on SR as well, so I'm shit out of luck.

> not really a fan? The hardsuits are iconic. I feel like its too easy to confuse them with the other white hardsuits we currently have (especially the security one that doesn't see action most rounds)
> 
> Not to mention they aren't... y'know... _blood red_ anywhere. They _need_ to be iconic as the crew is trained to spot the originals and almost everyone who plays ss13 knows what a blood-red hardsuit normally looks like.

There are exactly two white hardsuits in game and in code, one being the security one as you said and the CE's suit. The reskin variant differentiates itself from both of them by having something extremely different, with the tassets on the legs and the bright red spine light in EVA mode, and even then the tassetless version has a different look compared to them in the spine light and how it's designed anyway.
The helmet also is still the same look as the second most iconic part of the hardsuit. It's 4 bright red eyes + antennae compared to both the CE's blue Isaac Clarke style slits and the Sec voidsuit's blue mono visor.

Comparisons between the three:

comparison



> They already decided to differentiate sec from the syndicate with red and blue, what justifies changing the syndicate red and black to a grey, white and black?

This isn't removing the original Redsuit nor changing anything else that's Black and red in the Syndicate, and it still has the motif of black and red, just a white, black and a hint of red motif instead. I'm just giving players a choice to look either winterized or original.

A ''''lore'''' friendly way to add it in is also "Yeah we have bases and operatives on winter planets. Let's roll out a suit to help with those conditions."

**Comment by Ebin-Halcyon** [comment]
> > why doesnt this just use the alternate skin system that like double barrel shotguns or captain pens have
> 
> Tried doing it before with Orion but couldn't get it to work, along with alt-click already being the 'open the cell cover' hotkey I'm a shoddy at best coder and wouldn't be able to make a work around or stuff in already existing code and make it work. Everyone that could help is a bit busy with making the new prefs menu work on SR as well, so I'm shit out of luck.
> 
> > not really a fan? The hardsuits are iconic. I feel like its too easy to confuse them with the other white hardsuits we currently have (especially the security one that doesn't see action most rounds)
> > Not to mention they aren't... y'know... _blood red_ anywhere. They _need_ to be iconic as the crew is trained to spot the originals and almost everyone who plays ss13 knows what a blood-red hardsuit normally looks like.
> 
> There are exactly two white hardsuits in game and in code, one being the security one as you said and the CE's suit. The reskin variant differentiates itself from both of them by having something extremely different, with the tassets on the legs and the bright red spine light in EVA mode, and even then the tassetless version has a different look compared to them in the spine light and how it's designed anyway.
> The helmet also is still the same look as the second most iconic part of the hardsuit, It's 4 bright red eyes + antennae compared to both the CE's blue Isaac Clarke style slits and the Sec voidsuit's blue mono visor.
> 
> Comparisons between the three:
> 
> comparison
> 
> > They already decided to differentiate sec from the syndicate with red and blue, what justifies changing the syndicate red and black to a grey, white and black?
> 
> This isn't removing the original Redsuit nor changing anything else that's Black and red in the Syndicate, and it still has the motif of black and red, just a white, black and a hint of red motif instead. I'm just giving players a choice to look either winterized or original.
> 
> A ''''lore'''' friendly way to add it in is also "Yeah we have bases and operatives on winter planets. Let's roll out a suit to help with those conditions."

**Comment by YakumoChen** [comment]
The sprites are not distinctively different aside from the little lights on them and the shape of the helmet.

Again, players are trained, both mechanically and through experience, to report _blood red_ hardsuits. These I feel should be limited to say, interdyne and not general antagonists.

**Comment by ghost** [comment]
> why doesnt this just use the alternate skin system that like double barrel shotguns or captain pens have

To be more specific than Halcyons answer, me and them tried this. Its coded with single objects in mind, the suit-helmet combo is two objects. To top it off, all the suit toggling stuff is stored in the helmet, while the altskin stuff would need to be stored in the suit itself (as the helmet isnt always accessible)
It was janky and gross and didnt work regardless; tldr altskin is for single obj, hardsuits are two objs

**Comment by Ebin-Halcyon** [comment]
> The sprites are not distinctively different aside from the little lights on them and the shape of the helmet.
> 
> Again, players are trained, both mechanically and through experience, to report _blood red_ hardsuits. These I feel should be limited to say, interdyne and not general antagonists.

Fair I suppose, i'll just make it Interdyne DS-2 Specific.

**Comment by axietheaxolotl** [comment]
based and syndiepilled

**Comment by Ebin-Halcyon** [comment]
@GoldenAlpharex Mind taking a glance over these and seeing if it's all fine? I should be finished with just about everything.

**Comment by GoldenAlpharex** [comment]
> @GoldenAlpharex Mind taking a glance over these and seeing if it's all fine? I should be finished with just about everything.

I'm dumb and didn't get the message, hit me up on Discord next time.

**Comment by Ebin-Halcyon** [comment]
> > @GoldenAlpharex Mind taking a glance over these and seeing if it's all fine? I should be finished with just about everything.
> 
> I'm dumb and didn't get the message, hit me up on Discord next time.

Gotcha

File: modular_skyrat/modules/Winter-hardsuit/code/clothing/suit/winter_hardsuit.dm [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/obj/item/clothing/head/helmet/space/hardsuit/syndi/winter
+	name = "Winterized Syndicate hardsuit helmet"
```
Comment (axietheaxolotl) [inline modular_skyrat/modules/Winter-hardsuit/code/clothing/suit/winter_hardsuit.dm]: yoyo

File: modular_skyrat/modules/Winter-hardsuit/code/clothing/suit/winter_hardsuit.dm [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	desc = "A winterized variant of the ever feared Blood-Red Hardsuit for work in frozen conditions, despite the extra padding for comfort it does not have superior armor, it is in EVA mode. Property of Gorlex Marauders."
+	alt_desc = "A winterized variant of the ever feared Blood-Red Hardsuit for work in frozen conditions, despite the extra padding for comfort it does not have superior armor, it is in combat mode. Property of Gorlex Marauders."
+	icon = 'modular_skyrat/master_files/icons/obj/clothing/hats.dmi'
+	worn_icon = 'modular_skyrat/master_files/icons/mob/clothing/head.dmi'
+	icon_state = "hardsuit1-syndi-winter"
+	inhand_icon_state = "syndie_helm"
+	hardsuit_type = "syndi-winter"
+
+/obj/item/clothing/suit/space/hardsuit/syndi/winter
+	name = "Winterized Syndicate Hardsuit"
```
Comment (axietheaxolotl) [inline modular_skyrat/modules/Winter-hardsuit/code/clothing/suit/winter_hardsuit.dm]: Same here.

File: modular_skyrat/modules/Winter-hardsuit/code/clothing/suit/winter_hardsuit.dm [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	icon = 'modular_skyrat/master_files/icons/obj/clothing/suits.dmi'
+	worn_icon = 'modular_skyrat/master_files/icons/mob/clothing/suit.dmi'
+	icon_state = "hardsuit1-syndi-winter"
+	inhand_icon_state = "syndie_hardsuit"
+	hardsuit_type = "syndi-winter"
+	helmettype = /obj/item/clothing/head/helmet/space/hardsuit/syndi/winter
+	hardsuit_tail_colors = list("EEE", "111", "211")
+
+/obj/item/clothing/head/helmet/space/hardsuit/syndi/wintertas
+	name = "Winterized Syndicate hardsuit helmet"
```
Comment (axietheaxolotl) [inline modular_skyrat/modules/Winter-hardsuit/code/clothing/suit/winter_hardsuit.dm]: weh

File: modular_skyrat/modules/Winter-hardsuit/code/clothing/suit/winter_hardsuit.dm [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+	desc = "A winterized variant of the ever feared Blood-Red Hardsuit for work in frozen conditions, despite the extra padding for comfort it does not have superior armor, it is in EVA mode. Property of Gorlex Marauders."
+	alt_desc = "A winterized variant of the ever feared Blood-Red Hardsuit for work in frozen conditions, despite the extra padding for comfort it does not have superior armor, it is in combat mode. Property of Gorlex Marauders."
+	icon = 'modular_skyrat/master_files/icons/obj/clothing/hats.dmi'
+	worn_icon = 'modular_skyrat/master_files/icons/mob/clothing/head.dmi'
+	icon_state = "hardsuit1-syndi-wintertas"
+	inhand_icon_state = "syndie_helm"
+	hardsuit_type = "syndi-wintertas"
+
+/obj/item/clothing/suit/space/hardsuit/syndi/wintertas
+	name = "Winterized Syndicate Hardsuit"
```
Comment (axietheaxolotl) [inline modular_skyrat/modules/Winter-hardsuit/code/clothing/suit/winter_hardsuit.dm]: guh

**Review by axietheaxolotl — CHANGES_REQUESTED** [review]
Don't start non-proper nouns (Winterized) with capitals in item names, it looks fucky in game

**Pull request merged**